python/shark_turbine/dynamo/importer.py

```diff
@@ -617,6 +647,26 @@ def _make_constant_op(
         attributes={"value": value_attr},
     )
 
+def _make_vtensor_literal_op(tensor: torch.Tensor, mlir_type: MlirType) -> Operation:
+    npy_dtype = TORCH_DTYPE_TO_NPY_TYPE.get(tensor.dtype)
+    assert npy_dtype is not None, f"Can not create literal tensor for unsupported datatype: {tensor.dtype}"
+    # We need a raw buffer of data in order to create an ElementsAttr for the invocation of torch.vtensor.literal,
+    # but torch.Tensor does not fulfill the python buffer/array interface hence we must convert to a numpy array to get
+    # a raw buffer of our data. We can't call torch.Tensor.numpy() directly because this internally forces a call to
+    # detach() which throws an error as we are operating in a FakeTensorMode, hence the simplest way to get this raw
+    # buffer is via the indirection: Tensor -> list -> numpy array. This allows us to create a vtensor literal as
+    # desired, but also limits which data types we can support in this function (see TORCH_DTYPE_TO_NPY_TYPE above)
+    np_tensor = np.array(tensor.tolist()).astype(npy_dtype)
```

Comment:
Oof. This is going to come back to bite us. Would you mind capturing the context here in an issue and noting it in the PR description to tie together? This kind of stuff is always like walking on broken glass a bit and we'll need to revisit at some point.

Reply:
Sounds good I'll write it up tomorrow morning, I wish there was a better way to get the buffer representation of the tensor but so far there is not -  if someone were to finish implementing the python array interface for tensors in pytorch it would give us a first-class mechanism for getting the buffer representation and we could avoid this.

This is an issue tracking this feature: https://github.com/pytorch/pytorch/issues/54138

Alternatively, we could manually do some sort of restructuring from numpy format to the elementsattr buffer format, not sure where to find those docs at the moment though, could be worth finding them.